build/log.go

```diff
@@ -50,8 +51,8 @@ type LogWriter struct {
 // NewSubLogger constructs a new subsystem log from the current LogWriter
 // implementation. This is primarily intended for use with stdlog, as the actual
 // writer is shared amongst all instantiations.
-func NewSubLogger(subsystem string,
-	genSubLogger func(string) btclog.Logger) btclog.Logger {
+func NewSubLogger(subsystem string, interceptor signal.Interceptor,
```

Comment:
Can we get away with not needing to pass the interceptor in here and leave that the the `genSubLogger` function? That way we don't need to pass all the empty interceptors in. 

So the signature here is: 
`NewSubLogger(subsystem string, genSubLogger func(string, func()) btclog.Logger)`

When we don't have access to an interceptor we can just call it with:
`build.NewSublogger(subsystem, nil)`

When we do have access to an interceptor, we use:
```
shutdown:=func(){
  if !interceptor.Listening() {
    return
  }

  interceptor.RequestShutdown()
}

build.NewSublogger(subsystem, shutdown)
```

Tried this out quickly [here](https://github.com/carlaKC/lnd/commit/abfce52064fe5d1e684e81d5a2f7e64fb0ffb56c) and it seems to work, although the code could probably be a bit cleaner 😅

Reply:
Thanks, I cherry-picked your commit. It is much better.